pkg/store/bucket.go

```diff
@@ -325,16 +314,26 @@ func (s *BucketStore) getBlock(id ulid.ULID) *bucketBlock {
 func (s *BucketStore) setBlock(id ulid.ULID, b *bucketBlock) {
 	s.mtx.Lock()
 	defer s.mtx.Unlock()
-
 	s.blocks[id] = b
+}
 
-	if s.oldestBlockMinTime > b.meta.MinTime || s.oldestBlockMinTime == math.MaxInt64 {
-		s.oldestBlockMinTime = b.meta.MinTime
-	}
-	if s.youngestBlockMaxTime < b.meta.MaxTime || s.youngestBlockMaxTime == math.MaxInt64 {
-		s.youngestBlockMaxTime = b.meta.MaxTime
+// TimeRange returns the minimum and maximum timestamp of data available in the store.
+func (s *BucketStore) TimeRange() (mint, maxt int64) {
```

Comment:
I wanted to do that in the same loop... But I guess this is more clear way. (: sgtm

Reply:
Yea, at the point where we started removing blocks the previous code was essentially no longer correct. And just re-computing it on every call is still cheap and a lot easier than handling the removal case.